### PR TITLE
User SecurityStamp validation added

### PR DESCRIPTION
**Pull request by andriikaplanovskyi**
Added user SecurityStamp validation to expire all existing authentication cookies for a user if they logged out.

### CODE REVIEW

**Comment by andriikaplanovskyi** [comment]
@fabiomaistro please do not complete until Wednesday, as @dcsl-alexis needs this issue on QA for demo purposes.

File: src/OneBeyond.Studio.Obelisk.Authentication/OneBeyond.Studio.Obelisk.Authentication.Application/DependencyInjection/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -93,6 +93,12 @@ public static class ServiceCollectionExtensions
         services.AddSingleton((_) =>
             configuration.GetOptions<CookieAuthNOptions>("CookieAuthN"));
 
+        services.Configure<SecurityStampValidatorOptions>(options =>
+        {
+            //Every time we logout we update the user security stamp to make sure that all existing auth cookies are invalidated
+            options.ValidationInterval = TimeSpan.FromSeconds(10);
```
Comment (fabiomaistro) [inline src/OneBeyond.Studio.Obelisk.Authentication/OneBeyond.Studio.Obelisk.Authentication.Application/DependencyInjection/ServiceCollectionExtensions.cs:99]: I'm not sure on why we are changing this, can you explain @andrii-kaplanovskyi ?

**Review by fabiomaistro — CHANGES_REQUESTED** [review]

File: src/OneBeyond.Studio.Obelisk.Authentication/OneBeyond.Studio.Obelisk.Authentication.Application/DependencyInjection/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (andriikaplanovskyi) [reply src/OneBeyond.Studio.Obelisk.Authentication/OneBeyond.Studio.Obelisk.Authentication.Application/DependencyInjection/ServiceCollectionExtensions.cs:99]: The default validation interval is 30 minutes. We need to do it (validate security stamps) quicker.

File: src/OneBeyond.Studio.Obelisk.Authentication/OneBeyond.Studio.Obelisk.Authentication.Application/DependencyInjection/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -93,6 +93,12 @@ public static class ServiceCollectionExtensions
         services.AddSingleton((_) =>
             configuration.GetOptions<CookieAuthNOptions>("CookieAuthN"));
 
+        services.Configure<SecurityStampValidatorOptions>(options =>
+        {
+            //Every time we logout we update the user security stamp to make sure that all existing auth cookies are invalidated
```
Comment (fabiomaistro) [inline src/OneBeyond.Studio.Obelisk.Authentication/OneBeyond.Studio.Obelisk.Authentication.Application/DependencyInjection/ServiceCollectionExtensions.cs:98]: I would put a different comment saying:
`// Decrease the default interval between security stamp validations from 30 minutes to 1 minute`

**Review by fabiomaistro — CHANGES_REQUESTED** [review]

**Review by fabiomaistro — APPROVED** [review]

**Pull request merged**